src/node_internals.h

```diff
@@ -75,6 +75,100 @@ struct sockaddr;
+    V(cares_wrap,     MODREG_WITH_DEF)                                        \
+    V(fs_event_wrap,  MODREG_WITH_DEF)                                        \
+    V(js_stream,      MODREG_WITH_DEF)                                        \
+    V(module_wrap,    MODREG_WITH_DEF)                                        \
+    V(buffer,         MODREG_WITH_DEF)                                        \
+    V(config,         MODREG_WITH_DEF)                                        \
+    V(contextify,     MODREG_WITH_DEF)                                        \
+    V(fs,             MODREG_WITH_DEF)                                        \
+    V(http2,          MODREG_WITH_DEF)                                        \
+    V(http_parser,    MODREG_WITH_DEF)                                        \
+    V(os,             MODREG_WITH_DEF)                                        \
+    V(performance,    MODREG_WITH_DEF)                                        \
+    V(serdes,         MODREG_WITH_DEF)                                        \
+    V(url,            MODREG_WITH_DEF)                                        \
+    V(util,           MODREG_WITH_DEF)                                        \
+    V(v8,             MODREG_WITH_DEF)                                        \
+    V(zlib,           MODREG_WITH_DEF)                                        \
+    V(icu,            ICU_MODREG_DEF)                                         \
+    V(pipe_wrap,      MODREG_WITH_DEF)                                        \
+    V(process_wrap,   MODREG_WITH_DEF)                                        \
+    V(signal_wrap,    MODREG_WITH_DEF)                                        \
+    V(spawn_sync,     MODREG_WITH_DEF)                                        \
+    V(stream_wrap,    MODREG_WITH_DEF)                                        \
+    V(tcp_wrap,       MODREG_WITH_DEF)                                        \
+    V(timer_wrap,     MODREG_WITH_DEF)                                        \
+    V(tty_wrap,       MODREG_WITH_DEF)                                        \
+    V(udp_wrap,       MODREG_WITH_DEF)                                        \
+    V(uv,             MODREG_WITH_DEF)                                        \
+    V(inspector,      MODREG_WITH_DEF)                                        \
+    V(crypto,         OPENSSL_MODREG_DEF)                                     \
+    V(tls_wrap,       OPENSSL_MODREG_DEF)
+
+/* for nodejs built-in modules only */
+#define NODE_MODULE_CONTEXT_AWARE_X_BUILTIN(modname,                          \
+                                            regfunc,                          \
+                                            priv,                             \
+                                            flags)                            \
+  static node::node_module _module =                                          \
+  {                                                                           \
```

Comment:
Style nit: `{` should go on the previous line and since you're here, can you replace `NULL` with `nullptr`?